examples/xrf_gui/xrf_gui.py

```diff
@@ -1,20 +1,115 @@
+# Redistribution and use in source and binary forms, with or without   #
+# modification, are permitted provided that the following conditions   #
+# are met:                                                             #
+#                                                                      #
+# * Redistributions of source code must retain the above copyright     #
+#   notice, this list of conditions and the following disclaimer.      #
+#                                                                      #
+# * Redistributions in binary form must reproduce the above copyright  #
+#   notice this list of conditions and the following disclaimer in     #
+#   the documentation and/or other materials provided with the         #
+#   distribution.                                                      #
+#                                                                      #
+# * Neither the name of the Brookhaven Science Associates, Brookhaven  #
+#   National Laboratory nor the names of its contributors may be used  #
+#   to endorse or promote products derived from this software without  #
+#   specific prior written permission.                                 #
+#                                                                      #
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS  #
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT    #
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS    #
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE       #
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,           #
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES   #
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR   #
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   #
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  #
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
 
+__author__ = 'Li Li'
+
+import os
+import numpy as np
 import enaml
 from enaml.qt.qt_application import QtApplication
-from bubblegum.xrf.model.xrf_model import XRF
+from bubblegum.xrf.model.fileio import FileIOModel
+from bubblegum.xrf.model.lineplot import LinePlotModel
+from bubblegum.xrf.model.guessparam import GuessParamModel
+from bubblegum.xrf.model.draw_image import DrawImage
+from bubblegum.xrf.model.fit_spectrum import Fit1D, Param
+from bubblegum.xrf.model.setting import SettingModel
+import json
+from pprint import pprint
+
+
+def get_defaults():
+
+    #working_directory = os.path.join(os.path.expanduser('~'), 'Downloads')
+    working_directory = '/Users/Li/Research/X-ray/Research_work/all_code/nsls2_gui/nsls2_gui'
```

Comment:
The IO part is poor organized, because it will be replaced by file store and data muggler eventually. I will correct the path, so others can easily run test.